impeller/entity/contents/content_context.cc

```diff
@@ -604,4 +607,70 @@ void ContentContext::FlushCommandBuffers() const {
   GetContext()->GetCommandQueue()->Submit(buffers);
 }
 
+void ContentContext::InitializeCommonlyUsedShadersIfNeeded() const {
+  TRACE_EVENT0("flutter", "InitializeCommonlyUsedShadersIfNeeded");
+
+  // Initialize commonly used shaders that aren't defaults. These settings were
+  // chosen based on the knowledge that we mix and match triangle and
+  // triangle-strip geometry, and also have fairly agressive srcOver to src
+  // blend mode conversions.
+  auto options = ContentContextOptions{
+      .sample_count = SampleCount::kCount4,
+      .color_attachment_pixel_format =
+          context_->GetCapabilities()->GetDefaultColorFormat()};
+
+  for (const auto mode : {BlendMode::kSource, BlendMode::kSourceOver}) {
+    for (const auto geometry :
+         {PrimitiveType::kTriangle, PrimitiveType::kTriangleStrip}) {
+      options.blend_mode = mode;
+      options.primitive_type = geometry;
+      CreateIfNeeded(solid_fill_pipelines_, options);
+      if (context_->GetCapabilities()->SupportsSSBO()) {
+        CreateIfNeeded(linear_gradient_ssbo_fill_pipelines_, options);
+        CreateIfNeeded(radial_gradient_ssbo_fill_pipelines_, options);
+        CreateIfNeeded(sweep_gradient_ssbo_fill_pipelines_, options);
+        CreateIfNeeded(conical_gradient_ssbo_fill_pipelines_, options);
+        CreateIfNeeded(texture_pipelines_, options);
+      }
+    }
+  }
+
+  options.blend_mode = BlendMode::kDestination;
+  options.primitive_type = PrimitiveType::kTriangleStrip;
+  for (const auto stencil_mode :
+       {ContentContextOptions::StencilMode::kLegacyClipIncrement,
+        ContentContextOptions::StencilMode::kLegacyClipDecrement,
+        ContentContextOptions::StencilMode::kLegacyClipRestore}) {
+    options.stencil_mode = stencil_mode;
+    CreateIfNeeded(clip_pipelines_, options);
+  }
+
+  if (GetContext()->GetBackendType() == Context::BackendType::kVulkan) {
```

Comment:
@chinmaygarde where did we land on making this vulkan only? IIRC we agreed not to do vendor checks since this code should be harmless.

Reply:
I removed the Vulkan only check since it makes testing more of a pain.